Mypy is failing:

```

task-sdk/src/airflow/sdk/execution_time/supervisor.py:1580: error: Incompatible
types in assignment (expression has type "TextIOWrapper", variable has type
"Optional[BufferedWriter]")  [assignment]
                log_file_descriptor = log_file.open("a", buffering=1)
                                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
task-sdk/src/airflow/sdk/execution_time/supervisor.py:1581: error: Argument 1
to "WriteLogger" has incompatible type "Optional[BufferedWriter]"; expected
"Optional[TextIO]"  [arg-type]
    ...ing_logger: WrappedLogger = structlog.WriteLogger(log_file_descriptor)
                                                         ^~~~~~~~~~~~~~~~~~~
Found 2 errors in 1 file (checked 110 source files)

Error 1 returned
```

--- a/task-sdk/src/airflow/sdk/execution_time/supervisor.py
+++ b/task-sdk/src/airflow/sdk/execution_time/supervisor.py
@@ -32,10 +32,10 @@ from collections.abc import Generator
 from contextlib import contextmanager, suppress
 from datetime import datetime, timezone
 from http import HTTPStatus
-from io import BufferedWriter
 from socket import SO_SNDBUF, SOL_SOCKET, SocketIO, socket, socketpair
 from typing import (
     TYPE_CHECKING,
+    BinaryIO,
     Callable,
     ClassVar,
     NoReturn,
@@ -1566,7 +1566,7 @@ def supervise(
 
     # TODO: Use logging providers to handle the chunked upload for us etc.
     logger: FilteringBoundLogger | None = None
-    log_file_descriptor: BufferedWriter | None = None
+    log_file_descriptor: BinaryIO | TextIO | None = None
     if log_path:
         # If we are told to write logs to a file, redirect the task logger to it. Make sure we append to the
         # file though, otherwise when we resume we would lose the logs from the start->deferral segment if it
@@ -1577,10 +1577,10 @@ def supervise(
 
         pretty_logs = False
         if pretty_logs:
-            log_file_descriptor = log_file.open("a", buffering=1)
+            log_file_descriptor: TextIO = log_file.open("a", buffering=1)
             underlying_logger: WrappedLogger = structlog.WriteLogger(log_file_descriptor)
         else:
-            log_file_descriptor = log_file.open("ab")
+            log_file_descriptor: BinaryIO = log_file.open("ab")
             underlying_logger = structlog.BytesLogger(log_file_descriptor)
         processors = logging_processors(enable_pretty_log=pretty_logs)[0]
         logger = structlog.wrap_logger(underlying_logger, processors=processors, logger_name="task").bind()

--- a/task-sdk/src/airflow/sdk/execution_time/supervisor.py
+++ b/task-sdk/src/airflow/sdk/execution_time/supervisor.py
@@ -1608,5 +1608,4 @@ def supervise(
     log.info("Task finished", exit_code=exit_code, duration=end - start, final_state=process.final_state)
     if log_path and log_file_descriptor:
         log_file_descriptor.close()
-        log.info("Log file closed successfully", log_path=log_path)
     return exit_code

--- a/task-sdk/src/airflow/sdk/execution_time/supervisor.py
+++ b/task-sdk/src/airflow/sdk/execution_time/supervisor.py
@@ -1577,11 +1577,11 @@ def supervise(
 
         pretty_logs = False
         if pretty_logs:
-            log_file_descriptor: TextIO = log_file.open("a", buffering=1)
-            underlying_logger: WrappedLogger = structlog.WriteLogger(log_file_descriptor)
+            log_file_descriptor = log_file.open("a", buffering=1)
+            underlying_logger: WrappedLogger = structlog.WriteLogger(cast(TextIO, log_file_descriptor))
         else:
-            log_file_descriptor: BinaryIO = log_file.open("ab")
-            underlying_logger = structlog.BytesLogger(log_file_descriptor)
+            log_file_descriptor = log_file.open("ab")
+            underlying_logger = structlog.BytesLogger(cast(BinaryIO, log_file_descriptor))
         processors = logging_processors(enable_pretty_log=pretty_logs)[0]
         logger = structlog.wrap_logger(underlying_logger, processors=processors, logger_name="task").bind()
 

